python_modules/libraries/dagster-databricks/dagster_databricks/databricks_pyspark_step_launcher.py

```diff
@@ -222,12 +222,17 @@ def launch_step(self, step_context):
         if self.permissions:
             self._grant_permissions(log, databricks_run_id)
 
+        completed = False
         try:
             # If this is being called within a `capture_interrupts` context, allow interrupts while
             # waiting for the  execution to complete, so that we can terminate slow or hanging steps
             with raise_execution_interrupts():
                 yield from self.step_events_iterator(step_context, step_key, databricks_run_id)
+            completed = True
         finally:
+            # if execution is interrupted before the step is completed, cancel the run
+            if not completed:
+                self.databricks_runner.client.client.jobs.cancel_run(databricks_run_id)
```

Comment:
is this meaningfully different from

```
except:
   self.databricks_runner.client.client.jobs.cancel_run(databricks_run_id)  
finally:
```

without the completed boolean?

Reply:
yeah that's a lot cleaner, so I switched to that (+ the raise so that the error isn't swallowed). Verified that cancelling a long-running step from the UI does indeed cancel the job on the databricks side.